From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@google.com>
Date: Fri, 18 Dec 2020 18:17:27 +0100
Subject: ANDROID: Incremental fs: switch ksys_close() to close_fd()

In commit 1572bfdf21d4 ("file: Replace ksys_close with close_fd"),
ksys_close is removed from the kernel, as the same functionality is now
in close_fd().  So do this for incfs so that the code continues to work
properly.

Fixes: 1572bfdf21d4 ("file: Replace ksys_close with close_fd")
Cc: Paul Lawrence <paullawrence@google.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@google.com>
Change-Id: I91aa9b6dfce2e09d5c16c0dce20fdfa6ead4009c
---
 fs/incfs/pseudo_files.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/fs/incfs/pseudo_files.c b/fs/incfs/pseudo_files.c
--- a/fs/incfs/pseudo_files.c
+++ b/fs/incfs/pseudo_files.c
@@ -8,6 +8,7 @@
 #include <linux/namei.h>
 #include <linux/poll.h>
 #include <linux/syscalls.h>
+#include <linux/fdtable.h>
 
 #include <uapi/linux/incrementalfs.h>
 
@@ -403,7 +404,7 @@ static int dir_relative_path_resolve(
 		LOOKUP_FOLLOW | LOOKUP_DIRECTORY, result_path, NULL);
 
 out:
-	ksys_close(dir_fd);
+	close_fd(dir_fd);
 	if (error)
 		pr_debug("incfs: %s %d\n", __func__, error);
 	return error;
